src/BreakfastSerializer/JSONSerializer.php

```diff
@@ -191,4 +202,31 @@ protected function objectToArray($baseObject, $exposeClassName = true)
         return $data;
     }
 
+    /**
+     * @param array $objAsArray
+     * @param array $data
+     * @param       $exposeClassName
+     * @param       $baseObject
+     * @param       $currentClassName
+     */
+    protected function iterateClassProperties(
+        array $objAsArray,
+        array& $data,
+        $exposeClassName,
+        $baseObject,
+        $currentClassName
+    )
```

Comment:
Aye, however, I would rather pass these params in and have smaller chunks of code, I've marked this as internal for now and have added a todo for when I do the next supported type to see how this can be made even smaller